### PR TITLE
Eager loading specs

### PR DESCRIPTION
**Pull request by fullstop125**
In the following milestone i managed to implement the following:  
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe view

#### Integration specs
-  Use Capybara to write integration tests for each view in your project.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -50,12 +50,17 @@ gem 'bootsnap', require: false
```
Comment (HAMMAS-SALEEM) [inline Gemfile:50]: Capture0

- [x] You have a link of your license file in your readme file 😊 but it is broken 😨 Kindly fix this link to make your readme file descriptive and license file easily accessible to all viewers 👍

Comment (HAMMAS-SALEEM) [inline Gemfile:50]: Capture

- [x] It's great that you have a license file in your project but it looks like a template 🤔 Kindly add your **name**, **current year** and **app-type** 😊 to make your project licensed to you 👍

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @fullstop125 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @fullstop125 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**